### PR TITLE
copied trivial from cats.core

### PR DESCRIPTION
**Pull request by kailuowang**

### CODE REVIEW

File: core/src/main/scala/cats/derived/trivial.scala [hunk 1]
```diff
@@ -20,3 +20,27 @@ trait Trivial1[F[_]]
...
+ * companion object, covering a few common use cases and avoiding the
+ * need for unnecessary lambdas (e.g. if you want a trivial type class
+ * instance for a type constructor, you should use `Trivial.PH1`).
+ *
+ * Moved from cats.core
+ */
+sealed trait Trivial
+
+object Trivial {
+  type P1[A] = Trivial
```
Comment (milessabin) [inline core/src/main/scala/cats/derived/trivial.scala]: What's the motivation for these aliases?

File: core/src/main/scala/cats/derived/trivial.scala [hunk 2]
```diff
@@ -20,3 +20,27 @@ trait Trivial1[F[_]]
...
+  type PH1[F[_]] = Trivial
+  type P1H1[F[_], A] = Trivial
+  type P2[A, B] = Trivial
+  type P2H1[F[_], A, B] = Trivial
+  type P3[A, B, C] = Trivial
+  type P3H1[F[_], A, B, C] = Trivial
+
+  implicit val catsTrivialInstance: Trivial = new Trivial {}
```
Comment (milessabin) [inline core/src/main/scala/cats/derived/trivial.scala]: I think `mkTrivial` would be better here, for consistency with `Trivial1`.

**Comment by milessabin** [comment]
OK, looking more closely, `Trivial.PH1` replaces `Trivial1` ... can we motivate the naming of the aliases and ditch `Trivial1` everywhere?

**Comment by kailuowang** [comment]
I replaced Trivial1 and renamed the instance creation method. By "motivate the naming of the aliases", did you mean naming them in a way that shows motivation?

**Comment by milessabin** [comment]
Well I'd like to know what the current set of names are meant to convey ... there's a pattern there, but it's not clear what it is or what the choice of letters signifies.

**Comment by kailuowang** [comment]
I had the same feeling, but I struggle to come up with a naming scheme myself. pinging @djspiewak

**Comment by djspiewak** [comment]
I'm guessing I meant "partial higher-order position arity-1", but that's just a guess. If the alias isn't obvious, it's a bad alias.

**Comment by kailuowang** [comment]
Closing due to lack of interest

**Pull request closed without merging**